docs/concepts/registries.md

```diff
@@ -40,63 +28,133 @@ public ExampleMod() {
 }
 ```
 
-Creating Registries
+### `Register` events
+
+The `RegistryEvent`s are the second and more flexible way to register objects. These [events][] are fired after the mod constructors and before the loading of configs.
+
+The event used in registering objects is the `RegistryEvent.Register<T>`. The type parameter `T` should be set to the type of the object being registered. Calling `#getRegistry` will return the registry, upon which objects are registered with `#register` or `#registerAll`. 
+
+`RegistryEvent.Register<?>` events are fired in this order: first, the `Block` registry, then the `Item` registry, and then all other registries in alphabetical order. 
+
+Here is an example: (note that the event handler must be registered on the *mod event bus* in the mod constructor)
+
+```java
+@SubscribeEvent
+public void registerBlocks(RegistryEvent.Register<Block> event) {
+    event.getRegistry().registerAll(block1, block2, ...);
+}
+```
+
+!!! note
+    `TileEntity` and `Entity` cannot be registered; instead, `TileEntityType` and `EntityType` are registered, and used in `TileEntity`/`Entity` constructors. These are created through the use of `TileEntityType.Builder` and `EntityType.Builder`, respectively. An example: (`REGISTER` refers to a `DeferredRegister<TileEntityType>`)
```

Comment:
Maybe specify ContainerType as well? Also could include that the "Type" classes are basically factories for the actual objects.

Reply:
Forgot about `ContainerType`, whoops. I'll include that, and that `*Type` classes are factories.